kombu/async/timer.py

```diff
@@ -41,7 +41,8 @@ def to_timestamp(d, default_timezone=utc, time=monotonic):
         if d.tzinfo is None:
             d = d.replace(tzinfo=default_timezone)
         diff = _time() - time()
-        return max((d - EPOCH).total_seconds() - diff, 0)
+        epoch = datetime.utcfromtimestamp(0).replace(tzinfo=default_timezone)
```

Comment:
```

In [22]: from pytz import UTC

In [23]: from pytz.reference import Eastern

In [24]: from datetime import datetime

In [25]: datetime.utcfromtimestamp(0).replace(tzinfo=Eastern) - datetime.utcfrom
    ...: timestamp(0).replace(tzinfo=UTC)
Out[25]: datetime.timedelta(0, 18000)
```

`replace`ing here means that `epoch` is _no longer_ the the "proper" epoch. 

The whole point of storing timezone info in datetimes is to do math without worrying about timezones... so here if you need timezone-aware datetimes, you could just do `epoch = datetime.utcfromtimestamp(0).replace(tzinfo=UTC)` to get the "timezone-aware" epoch.